File: app/src/main/java/com/rafamatias/nlp/data/entity/SentimentRequest.java
package com.rafamatias.nlp.data.entity;

public class SentimentRequest {
    private Document document;
    private final String encodingType = "UTF8";

    private SentimentRequest(Document document) {
        this.document = document;
    }

    static class Document {
        public Document(String content) {
            this.content = content;
        }

        private String type = "PLAIN_TEXT";
        private String content;
    }

    public static SentimentRequest withContext(String content){
        return new SentimentRequest(new Document(content));

    }
}

File: data/src/main/java/com/shrpereira/tweetsense/data/remote/sentiment/SentimentRequest.kt
package com.shrpereira.tweetsense.data.remote.sentiment

import com.google.gson.annotations.SerializedName

private const val ENCODING_TYPE = "UTF8"
private const val REQUEST_DATA_TYPE = "PLAIN_TEXT"

data class SentimentRequest(
	@SerializedName("document") val document: DocumentRequest,
	@SerializedName("encodingType") val encodingType: String = ENCODING_TYPE
)

data class DocumentRequest(
	@SerializedName("content") val content: String = "",
	@SerializedName("type") val type: String = REQUEST_DATA_TYPE
)